source/core.py

```diff
@@ -432,27 +432,37 @@ def handlePowerStatusChange(self):
 	log.debug("Initializing core pump")
 	class CorePump(wx.Timer):
 		"Checks the queues and executes functions."
+		_inNotify=False
 		def Notify(self):
-			global _isPumpPending
-			_isPumpPending = False
-			watchdog.alive()
+			if self._inNotify:
+				# #7104: Timers in wxPython 4 are now re-entrant. However, wx.Yield is not allowed to be called recursively.
+				# Therefore, stop this timer callback from ever being re-entered.
+				# Note that If code does try to request a pump from within this call, it will be detected and re-queued by restarting the timer.

```

Comment:
Since the flow here is hard to understand, and hard to debug, it would be good to be quite clear in this comment. After reading this I wonder how calling `wx.Yield` relates to requesting a core pump, and whether calling `wx.Yield` results in a core pump request that will be detected and requeued.

Reply:
wx.Yield is used to pump the window message queue, necessary when sending key presses, etc. However, that might end up processing the timer message queued by a previous pump request. We don't want that to happen, so we ignore it and re-queue it.

wx.Yield can also cause a pump request to be queued; e.g. if an accessibility event got handled during the yield.

All of that said, I'd suggest considering that this non-re-entrant timer code be abstracted, rather than handling it specifically for the core pump. There are other places where re-entrant timers will hurt us; e.g. updateCheck (see around line 276 of updateCheck.py).